pkg/cluster/registerWithHive.go

```diff
@@ -0,0 +1,127 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"os"
+	"time"
+
+	hive "github.com/Azure/ARO-RP/pkg/hiveregistrator"
+	"github.com/openshift/installer/pkg/asset/installconfig/azure"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (m *manager) collectDataForHive(ctx context.Context) error {
+
+	m.log.Print("collecting registration data from new cluster")
+
+	cv, err := m.configcli.ConfigV1().ClusterVersions().Get(ctx, "version", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	clusterid := cv.Spec.ClusterID
+
+	servicePrincipal := azure.Credentials{
+		TenantID:       m.env.TenantID(),
+		SubscriptionID: m.env.SubscriptionID(),
+		ClientID:       m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.ClientID,
+		ClientSecret:   string(m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.ClientSecret),
```

Comment:
Note: for adoption of existing clusters we need to grab it from the cluster as customer can update/rotate service principal.

We also need to firgure out whether we need to keep cluster SP in sync with what we have in hive. This is probably out of scope for this PR, but we need to think about it.

Reply:
Need to check: can we adopt cluster without providing service principals to Hive to avoid sync question?

Reply:
> Note: for adoption of existing clusters we need to grab it from the cluster as customer can update/rotate service principal.

I believe that the credentials we store in our database should be valid.  A customer should only rotate their service principal using `az aro update --refresh-credentials` which stores their service principal client id + client secret in our database.  I believe we don't have to grab them from the cluster.  

Additionally, our `m.validateResources` step validates that the cluster service principal (client id and client secret) is valid and has the correct permissions, which happens on `Update()` as well as `Install()`.

Reply:
> Note: for adoption of existing clusters we need to grab it from the cluster as customer can update/rotate service principal.

@m1kola
I believe that the credentials we store in our database should be valid.  A customer should only rotate their service principal using `az aro update --refresh-credentials` which stores their service principal client id + client secret in our database.  I believe we don't have to grab them from the cluster.  

Additionally, our `m.validateResources` step validates that the cluster service principal (client id and client secret) is valid and has the correct permissions, which happens on `Update()` as well as `Install()`.

Reply:
@bennerv ok, this solves the first bit (grabbing from the cluster), but we need to somehow keep SP stored in AKS alongside `ClusterDeployment` in sync with what we have in our DB and in the customer cluster.